go/felix/rules/dispatch_test.go

```diff
@@ -0,0 +1,350 @@
+					expDropRule,
+				},
+			},
+			{
+				Name: "cali-to-wl-dispatch",
+				Rules: []iptables.Rule{
+					outboundGotoRule("cali1234", "calitw-cali1234"),
+					expDropRule,
+				},
+			},
+		}),
+		Entry("interfaces sharing prefix", []string{"cali1234", "cali2333", "cali2444"}, []*iptables.Chain{
+			{
+				Name: "cali-from-wl-dispatch-2",
+				Rules: []iptables.Rule{
+					inboundGotoRule("cali2333", "califw-cali2333"),
+					inboundGotoRule("cali2444", "califw-cali2444"),
+					expDropRule,
+				},
+			},
+			{
+				Name: "cali-to-wl-dispatch-2",
+				Rules: []iptables.Rule{
+					outboundGotoRule("cali2333", "calitw-cali2333"),
+					outboundGotoRule("cali2444", "calitw-cali2444"),
+					expDropRule,
+				},
+			},
+			{
+				Name: "cali-from-wl-dispatch",
+				Rules: []iptables.Rule{
+					inboundGotoRule("cali1234", "califw-cali1234"),
+					inboundGotoRule("cali2+", "cali-from-wl-dispatch-2"),
+					expDropRule,
+				},
+			},
+			{
+				Name: "cali-to-wl-dispatch",
+				Rules: []iptables.Rule{
+					outboundGotoRule("cali1234", "calitw-cali1234"),
+					outboundGotoRule("cali2+", "cali-to-wl-dispatch-2"),
+					expDropRule,
+				},
+			},
+		}),
+		Entry("Multiple interfaces sharing multiple prefixes",
+			[]string{"cali11", "cali12", "cali13", "cali21", "cali22"},
+			[]*iptables.Chain{
+				{
+					Name: "cali-from-wl-dispatch-1",
+					Rules: []iptables.Rule{
+						inboundGotoRule("cali11", "califw-cali11"),
+						inboundGotoRule("cali12", "califw-cali12"),
+						inboundGotoRule("cali13", "califw-cali13"),
+						expDropRule,
+					},
+				},
+				{
+					Name: "cali-to-wl-dispatch-1",
+					Rules: []iptables.Rule{
+						outboundGotoRule("cali11", "calitw-cali11"),
+						outboundGotoRule("cali12", "calitw-cali12"),
+						outboundGotoRule("cali13", "calitw-cali13"),
+						expDropRule,
+					},
+				},
+				{
+					Name: "cali-from-wl-dispatch-2",
+					Rules: []iptables.Rule{
+						inboundGotoRule("cali21", "califw-cali21"),
+						inboundGotoRule("cali22", "califw-cali22"),
+						expDropRule,
+					},
+				},
+				{
+					Name: "cali-to-wl-dispatch-2",
+					Rules: []iptables.Rule{
+						outboundGotoRule("cali21", "calitw-cali21"),
+						outboundGotoRule("cali22", "calitw-cali22"),
+						expDropRule,
+					},
+				},
+				{
+					Name: "cali-from-wl-dispatch",
+					Rules: []iptables.Rule{
+						inboundGotoRule("cali1+", "cali-from-wl-dispatch-1"),
+						inboundGotoRule("cali2+", "cali-from-wl-dispatch-2"),
+						expDropRule,
+					},
+				},
+				{
+					Name: "cali-to-wl-dispatch",
+					Rules: []iptables.Rule{
+						outboundGotoRule("cali1+", "cali-to-wl-dispatch-1"),
+						outboundGotoRule("cali2+", "cali-to-wl-dispatch-2"),
+						expDropRule,
+					},
+				},
+			}),
+
+		// Duplicate interfaces could occur during transient misconfigurations, while
+		// there's no way to make them "work" since we can't distinguish the dupes, we
+		// should still render something sensible.
```

Comment:
Hm, I'm not sure the following output _is_ sensible.  Wonder how hard it would be to change the code so that duplicates like this are coalesced?